### PR TITLE
Add test for core implementation of XACML based scope validation for access token

### PR DESCRIPTION
**Pull request by senthalan**
depend on https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/801

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            + "<ns:Result>"
+            + "<ns:Decision>"
+            + DECISION
+            + "</ns:Decision>"
+            + "</ns:Result>"
+            + "</ns:root>";
+    private XACMLScopeValidator xacmlScopeValidator;
+    private AccessTokenDO accessTokenDO;
+    private OAuthAppDO authApp;
+    private Log log = mock(Log.class);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java]: Not required to mock the log class. Its handled separately outside the tests.

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    public void testCreateRequestDTO() throws Exception {
+
+        AuthenticatedUser authenticatedUser = mock(AuthenticatedUser.class);
+        when(accessTokenDO.getAuthzUser()).thenReturn(authenticatedUser);
+        when(accessTokenDO.getScope()).thenReturn(scopeArray);
+        when(authenticatedUser.getUserName()).thenReturn(ADMIN_USER);
+        when(authApp.getApplicationName()).thenReturn(APP_NAME);
+        RequestDTO requestDTO = WhiteboxImpl.invokeMethod(xacmlScopeValidator,
+                "createRequestDTO", accessTokenDO, authApp, resource);
+        assertTrue(requestDTO.getRowDTOs().size() == 9);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java]: Add a comment or error message to explain "9"

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            + "</ns:Decision>"
+            + "</ns:Result>"
+            + "</ns:root>";
+    private XACMLScopeValidator xacmlScopeValidator;
+    private AccessTokenDO accessTokenDO;
+    private OAuthAppDO authApp;
+    private String[] scopeArray;
+    private String resource;
+
+
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java]: remove extra new line

Comment (senthalan) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/XACMLScopeValidatorTest.java]: fixed with https://github.com/wso2-extensions/identity-oauth-addons/pull/22/commits/687f6b819361687150ed88fff5adc7d16f56d457

**Pull request merged**